apiserver/common/relationunitswatcher.go

```diff
@@ -9,28 +9,20 @@ import (
 	"github.com/juju/worker/v4/catacomb"
 
 	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/domain/relation"
 	"github.com/juju/juju/rpc/params"
-	"github.com/juju/juju/state"
 )
 
-// RelationUnitsWatcher represents a state.RelationUnitsWatcher at the
+// RelationUnitsWatcher represents a relation.RelationUnitsWatcher at the
 // apiserver level (different type for changes).
 type RelationUnitsWatcher interface {
 	watcher.Watcher[params.RelationUnitsChange]
-
-	// Stop is needed to implement facade.Resource.
-	// Deprecated: use Kill and Wait instead.
-	Stop() error
-
-	// Err implements watcher.Errer.
-	// Deprecated: use Kill and Wait instead.
-	Err() error
 }
 
-// RelationUnitsWatcherFromState wraps a state-level
+// RelationUnitsWatcherFromDomain wraps a state-level
 // RelationUnitsWatcher in an equivalent apiserver-level one, taking
 // responsibility for the source watcher's lifetime.
-func RelationUnitsWatcherFromState(source state.RelationUnitsWatcher) (RelationUnitsWatcher, error) {
+func RelationUnitsWatcherFromDomain(source relation.RelationUnitsWatcher) (RelationUnitsWatcher, error) {
```

Comment:
All this does is take a domain watcher and transform the received events to a different payload.
We could easily genericise it by implementing the base wrapper and passing in a mapper function. The base wrapper could live outside the facade packages and we'd by design get rid of this apiserver common artefact which is an end goal.